core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -87,9 +87,14 @@ class LogManager(logDirs: Array[File],
   @volatile private var logStartOffsetCheckpoints = liveLogDirs.map(dir =>
     (dir, new OffsetCheckpointFile(new File(dir, LogStartOffsetCheckpointFile), logDirFailureChannel))).toMap
 
-  private def offlineLogDirs = logDirs.filterNot(_liveLogDirs.contains)
   private val preferredLogDirs = new ConcurrentHashMap[TopicPartition, String]()
 
+  private def offlineLogDirs: Iterable[File] = {
```

Comment:
`contains` on a queue is O(n), so the previous code was O(n^2). I thought that was a bit surprising given the method name so I replaced it with a O(n) implementation. Probably won't make much of a difference unless people have a lot of log directories, but this could happen.